magenta-lib/src/main/scala/magenta/input/resolver/DeploymentGraphBuilder.scala

```diff
@@ -0,0 +1,46 @@
+package magenta.input.resolver
+
+import magenta.graph.{Edge, EndNode, Graph, MidNode, Node, StartNode}
+import magenta.input.Deployment
+
+object DeploymentGraphBuilder {
```

Comment:
I fail to see how this captures a dependency chain. Might need you to take me through it.

A depends on B, which depends on C, which depends on D. You only have three levels of node, so this would produce:

A ~> B
A ~> C
B ~> D
C ~> D

But not
B ~> C

I'm sure I've missed something.